Работа по 4 сапринту

AlexSunset opened the pull request.

Commit:
Работа по 4 сапринту

pomegranate282922 commented on src/main/java/page/object/MainPage.java:
```diff
@@ -0,0 +1,60 @@
+package page.object;
```
Тут получился пакет page, а внутри пакет object. Вряд ли это имелось в виду. Вместо точки используем нижнее подчеркивание в пакетах.

pomegranate282922 commented on src/test/java/FirstScenario.java:
```diff
@@ -0,0 +1,54 @@
+import static org.junit.Assert.assertEquals;
+
+@RunWith(Parameterized.class)
+public class FirstScenario {
```
Слишком общее название. Название должно отражать бизнес логику происходящего внутри. Нужно исправить

pomegranate282922 commented on src/test/java/FirstScenario.java:
```diff
@@ -0,0 +1,54 @@
+
+    @Parameterized.Parameters
+    public static Object[] getTextData() {
+        return new Object[][]{
```
Отлично! Уместно применена параметризация.

pomegranate282922 commented on src/test/java/FirstScenario.java:
```diff
@@ -0,0 +1,54 @@
+    @Test
+    public void answerDisplayedOnClickIsTrue() throws InterruptedException {
+        driver = new ChromeDriver();
+        driver.get("https://qa-scooter.praktikum-services.ru/");
```
Урлы лучше выносить как константы. Так их проще переиспользовать и изменять.

pomegranate282922 commented on src/test/java/FirstScenario.java:
```diff
@@ -0,0 +1,54 @@
+        MainPage mainPage = new MainPage(driver);
+        mainPage.scrollToQuestionList();
+        mainPage.questionElements().get(index).click();
+        Thread.sleep(1000);
```
Использовать слип - это последний шаг, когда уже ничего другого не помогает. Нужно заменить на явные ожидания определённых событий.
Нужно исправить

pomegranate282922 reviewed (commented):
Хорошая работа! Все нужные тесты реализованы, есть проверки, применена параметризация! Хорошо реализована POM.
Есть замечания к неймингу и использованию слипов - их нужно заменить на явные ожидания.
Остальное по мелочи.

Commit:
Исправления после ревью по работе 4 спринта:
-Изменил названия классов автотестов согласно бизнес логике
-Добавил неявные ожидания вместо слипов
-Изменил имя пакета
-Вынес url сайта в отдельную константу

pomegranate282922 reviewed (approved):
Отлично, все замечания исправлены!
